src/Language/PureScript/TypeChecker/Entailment.hs

```diff
@@ -266,6 +267,7 @@ entails SolverOptions{..} constraint context hints =
 
             canBeGeneralized :: Type -> Bool
             canBeGeneralized TUnknown{} = True
+            canBeGeneralized (KindedType t _) = canBeGeneralized t
```

Comment:
I tried disabling it and added this to the test file:

```purescript
main = do
  when (show (unit :: Unit :: *) == "unit") (log "Done")
  let show' x = show (x :: _ :: *)
  when (show' unit == "unit") (log "Done")
```

But the test still passed.  Just about to try:

```purescript
main = do
  when (show (unit :: Unit :: *) == "unit") (log "Done")
  when (show (unit :: _ :: *) == "unit") (log "Done")
```